pkg/daemon/discover/discover.go

```diff
@@ -51,19 +51,28 @@ var (
 	cmName          string
 	cm              *v1.ConfigMap
 	udevEventPeriod = time.Duration(5) * time.Second
+	useCVInventory  bool
 )
 
-func Run(context *clusterd.Context, probeInterval time.Duration) error {
+func Run(context *clusterd.Context, probeInterval time.Duration, useCV bool) error {
 	if context == nil {
 		return fmt.Errorf("nil context")
 	}
-	logger.Infof("device discovery interval is %s", probeInterval.String())
+	logger.Debugf("device discovery interval is %s", probeInterval.String())
+	logger.Debugf("use ceph-volume inventory is %t", useCV)
 	nodeName = os.Getenv(k8sutil.NodeNameEnvVar)
 	namespace = os.Getenv(k8sutil.PodNamespaceEnvVar)
 	cmName = k8sutil.TruncateNodeName(LocalDiskCMName, nodeName)
+	useCVInventory = useCV
 	sigc := make(chan os.Signal, 1)
 	signal.Notify(sigc, syscall.SIGTERM)
-	err := updateDeviceCM(context)
+
+	err := context.Clientset.CoreV1().ConfigMaps(namespace).Delete(cmName, &metav1.DeleteOptions{})
```

Comment:
Why do we delete the configmap? I thought we were just adding new ceph-volume properties to the existing configmap.

Reply:
Doing tests i realize that if you start to use this feature and no changes detected in the storage devices, the configmap was logically not updated and therefore the information coming from ceph volume was never included in the configmap.
Two options here. 
1. modify "checkDeviceListsEqual" 
In order to return false if we want to include ceph-volume info or not, and we have this info in the config map or not. 
I do not like this, because to have/not to have ceph-volume inventory info in the configmap does not imply real changes in the set of storage devices.

2. Clean the configmap when  we start the discovery daemon. 
It seems even healthy. :-). And to delete and write again the configmap is not a great overload (we always retrieve the current information about storage devices).

Reply:
Instead of deleting it, a more common k8s pattern is to update the resource if it already exists. So I would vote to update the configmap if the c-v data isn't there. I guess that means option 1.